### DISCUSSION TITLE
How to suppress operationId from openapi output?

### DISCUSSION THREAD

**Discussion by LadyCailin** [microsoft/typespec#3076] [Q&A] [answered]
My original openapi yaml doesn't use operationId, and populating it causes various downstream SDK generators to change their behavior, by changing the method name from whatever they default to, to what is now provided by TypeSpec. I would like to suppress the emission of it from the output, but I'm not sure how.

Expected output:
```
/:
    get:
      tags:
        - foo
      responses:
        '200':
          content:
            application/json:
              schema:
                type: array
                items:
                  type: string
```

Actual output:

```
  /:
    get:
      tags:
        - foo
      operationId: foo_root
      parameters: []
      responses:
        '200':
          description: The request has succeeded.
          content:
            application/json:
              schema:
                type: array
                items:
                  type: string
```

https://typespec.io/playground?c=aW1wb3J0ICJAdHlwZXNwZWMvaHR0cCI7DQp1c2luZyBUeXBlU3BlYy5IdHRwOw0KDQpAdGFnKCJmb28iKQ0KaW50ZXJmYWNlIGZvbyB7DQogIEByb3V0ZSgiL8QgICBAZ2V0IHJvb3QoKTogQXJyYXk8c3RyaW5nPjsNCn0%3D&e=%40typespec%2Fopenapi3&options=%7B%7D

I have also tried using `@operationId("")` but this just causes it to use the default of `foo_root` in this case. One option is of course to explicitly provide the value that would otherwise be the default for the downstream generators, but I would prefer to suppress this entirely.

**Comment by timotheeguerin** [comment, marked as answer]
I do not think it is possible, this is the first time we are getting ask this. We could consider adding this as an emitter option, can you file a feature request.
Do you have cases where you include the operation id or never? 

But for know I think you have to use the workaround you described name the operations in typespec directly with the names you would have had in your generated client.

**Reply by LadyCailin** [reply]
Actually, I just realized this is not really possible either, as the Java/client generator outputs one thing as default, and the TypeScript/server one outputs another as the default. So if I hardcode something, it will standardize the two sides, thus breaking one side or the other.  I can more easily refactor the server side though, so I think that's what I'll end up doing.

I never used the operation id, and always took whatever the default was from the generators.

**Reply by timotheeguerin** [reply]
created a feature request https://github.com/microsoft/typespec/issues/3080
